src/core/ReactCompositeComponent.js

```diff
@@ -123,7 +109,7 @@ var ReactCompositeComponentMixin = assign({},
     this._rootNodeID = null;
 
     this._instance.props = element.props;
-    this._instance.state = null;
+    // state gets set up during mount
```

Comment:
well it gets reset during mount. this can be simplified later. if it's undefined, it gets set to null.